subprojects/execution/src/main/java/org/gradle/internal/execution/history/impl/FileSystemSnapshotSerializer.java

```diff
@@ -221,4 +223,17 @@ public boolean isEmpty() {
             return stack.isEmpty();
         }
     }
+
+    private static String toAbsolutePath(Collection<String> parents, String fileName) {
+        int length = fileName.length() + parents.size()  + parents.stream()
+            .map(java.lang.String::length)
+            .reduce(0, Integer::sum);
```

Comment:
```suggestion
        int length = fileName.length() + parents.size()  + parents.stream().mapToInt(String::length).sum();
```

This can be simplified.

Reply:
Hah! I learned something today, thanks!